### PR TITLE
powerdns: fix 32-bit builds against glibc

### PR DESCRIPTION
**Pull request by vcunat**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] i686-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vcunat** [comment]
This is for `staging-next` because (surprisingly) the support for 64-bit time_t on 32-bit platforms was only added in glibc 2.34 which still isn't in nixpkgs master.

**Comment by vcunat** [comment]
The [announcement](https://sourceware.org/pipermail/libc-alpha/2021-August/129718.html) says
> full support requires a minimum kernel version of 5.1

but that's hopefully satisfied in most of the (32-bit) cases.

**Comment by NickCao** [comment]
What will happen if I run this on a pre 5.1 kernel?

**Comment by vcunat** [comment]
I have no idea beyond that announcement.  They say "full support", so I'd hope that with 5.0 you might not run into any problems even.

File: pkgs/servers/dns/pdns/default.nix [hunk 1]
```diff
@@ -20,6 +20,11 @@ stdenv.mkDerivation rec {
     libyamlcpp libsodium curl unixODBC openssl systemd lmdb tinycdb
   ];
 
+  # Configure phase requires 64-bit time_t even on 32-bit platforms.
+  NIX_CFLAGS_COMPILE = lib.optionals stdenv.hostPlaftorm.is32bit [
```
Comment (symphorien) [inline pkgs/servers/dns/pdns/default.nix:24]:
```suggestion
  NIX_CFLAGS_COMPILE = lib.optionals stdenv.hostPlatform.is32bit [
```

Comment (vcunat) [reply pkgs/servers/dns/pdns/default.nix]: Thanks, no idea how I missed this.

**Pull request merged**